### DISCUSSION TITLE
Cannot import clients to a federation server. It says the key is not valid. (Copying using the copy icon, not getting asterisks.)

### DISCUSSION THREAD

**Discussion by juanlib85** [utmstack/UTMStack#258] [Q&A]
[JJ Coker]

I tried it with two different UTMStack Master Servers I have up and running.  Both of their keys fail to import on the Federation server.  Not sure what to do.  I already have the individual servers pulling logs from their respective networks so they are working.  The Federation server is up and running on the internet with a true DNS name.

**Comment by osmontero** [comment]
Can you please send which versions of UTMStack and Federation server you are using?

**Reply by jjcoker09** [reply]
Sorry, this was sent over from the UTMStack support ticket system and I did not create it.  so...I didn't see a response.  Could you reopen this discussion, please?  The Master Servers are UTMStack version are 10.1.0 and 10.2.0.  Not sure the version of the Federation server, though I downloaded it and installed it when 10.1.0 was out for the UTMStack ISOs.  I installed Ubuntu 20.04 on a VM server and downloaded and installed the Federation server from github.  The website is up and running for it, but that is all I can say about it.

**Comment by osmontero** [comment]
@jdieguez89 can you please check this?

**Comment by jdieguez89** [comment]
Hi @jjcoker09, this is probably a compatibility issue, the Federation Service for v10 is published at [Federation Service Release v10](https://github.com/utmstack/UTMStackFSInstaller/releases/tag/v10.0.0), please install this version and let us know

**Comment by jjcoker09** [comment]
That worked!  It is taking the key now.

Unfortunately, it is saying that the client is unreachable.  I opened 22, 80, 443, 9000 and 50051 between the two systems and it still won’t connect.  Is there a special port that needs to be open between the master servers and the federation server?

Thanks,

JJ

**Comment by jdieguez89** [comment]
Hi @jjcoker09, The Federation service connects with the UTMStack panel API through HTTPS in port 443 and HTTP in port 80; the federation service instance must be able to connect to both ports; also, please make sure to add the client IP or hostname; notice that the protocol is not required.
We will add this information to the documentation since it is confusing; thank you.

![image](https://github.com/utmstack/UTMStack/assets/26327410/c6b937b6-7013-43f4-9915-100ca62b463a)
